mod/layout/lib/javascript/script_table.js.coffee

```diff
@@ -1,96 +1,77 @@
-# FIXME: change name to "toggle details"
+# FIXME: change name to "toggle_details"
 
 $(document).ready ->
+  $('body').on 'click', "[data-details-mark]", ->
+    (new decko.details).toggle $(this)
 
-  # details toggle
-  active_details = null
+  $('body').on 'click', ".details-close-icon", ->
+    (new decko.details).closeLast()
 
-  # hack to solve the problem that
-  # if you click somewhere in the details area
-  # it is closed because it is a child of .details-toggle
-  no_toggle = null
-  $('body').on 'click', ".details-toggle  .details", ->
-    no_toggle = true
+  $('body').on 'click', '.details ._update-details', (e) ->
+    (new decko.details).add $(this)
+    e.preventDefault()
 
-  $('body').on 'click', ".details-close-icon", ->
-    trToggleDetails this
+decko.details = (dSlot) ->
```

Comment:
a new practice to organize coffeescript?

Reply:
Taking an object oriented approach to decko.filter made things _much_ cleaner.  I went with the same approach this time, and I couldn't swear the rewards were as high, but I do find it easier to read.

The other nice thing is that when you do this (and put very little logic in the event handling), it's easier to test in the browser, because `decko` is global.